week-3-nestjs-back-end/w03e02-spot-the-bug/src/users/auth.service.ts

```diff
@@ -31,6 +31,16 @@ export class AuthService {
     return this.currentUser;
   }
 
+  async setCurrentUser(token: string): Promise<void> {
+    if (this.validTokens.length > 0) {
+      const val = this.validTokens.find((t) => t.value === token);
+
+      if (val) {
+        this.currentUser = await this.usersService.getOneById(val.userId);
```

Comment:
Po 1 - BRAWO 👍 Pięknie wychwycone gdzie jest `bug`

Rozwiązanie jest dobrym pomysłem ale ta końcówka - już nie.
Nie możemy w ogóle w `memory` albo inaczej w `service` przechowywać `currentUser` - bo co jeśli user nr1 wyśle po sobie 2 zapytania i to samo będzie z userem nr2 i załóżmy że te zapytania się jakoś "przetną" w czasie.

Wtedy tylko po 1 zapytaniu się wykona, bo pozostałe będą odrzucone (z racji na uwierzytelnienie)
Albo wyobraź sobie że `200` osób na raz pyta o jakieś zasoby.
Nie powinniśmy mieć ani tablicy userów, ani zapisanego pojedynczego w danym momencie.
To Twój `middleware` powinien odpakować token i powiązać z userem, ale przekazać go do `request` -  i tylko tam jest cykl życia usera, kto potrzebuje to stamtąd go bierze.